python/remotivelabs-broker/remotivelabs/broker/sync/client.py

```diff
@@ -0,0 +1,263 @@
+            return self.signal.integer
+        elif self.signal.HasField("double"):
+            return self.signal.double
+        elif self.signal.HasField("arbitration"):
+            return self.signal.arbitration
+        else:
+            return None
+
+    def timestamp_us(self):
+        return self.signal.timestamp
+
+    def name(self):
+        return self.signal.id.name
+
+    def namespace(self):
+        return self.signal.id.namespace.name
+
+    def value(self):
+        return self.__get_value()
+
+    def float_value(self):
+        v = self.__get_value()
+        if isinstance(v, float):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type float but got {type(v)}")
+
+    def int_value(self):
+        v = self.__get_value()
+        if isinstance(v, int):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type int but got {type(v)}")
+
+    def bool_value(self):
+        v = self.__get_value()
+        if isinstance(v, bool):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type bool but got {type(v)}")
+
+    def bytes_value(self):
+        v = self.__get_value()
+        if isinstance(v, bytes):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type bytes but got {type(v)}")
+
+    def as_dict(self):
+        return {
+            'timestamp_us': self.timestamp_us(),
+            'name': self.name(),
+            'namespace': self.namespace(),
+            'value': self.value()
+        }
+
+
+class SignalsInFrame(Iterable):
+
+    def __init__(self, signals: list[SignalWrapper]):
+        self.signals = signals
+        self.index = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            result = self.signals[self.index]
+        except IndexError:
+            raise StopIteration
+        self.index += 1
+        return result
+
+
+class SignalIdentifier:
+    def __init__(self, name: str, namespace: str):
+        self.name = name
+        self.namespace = namespace
+
+
+class BrokerException(Exception):
+    pass
+
+
+class Client:
+
+    def __init__(self, client_id: str = "broker_client"):
+
+        self._signal_creator = None
+        self._traffic_stub = None
+        self._system_stub = None
+        self._network_stub = None
+        self._intercept_channel = None
+        self.client_id = client_id
+
+        self.url = None
+        self.api_key = None
+        """Main function, checking arguments passed to script, setting up stubs, configuration and starting Threads."""
+        # Setting up stubs and configuration
+        self.on_connect: Union[Callable[[Client], None], None] = None
+        self.on_signals: Union[Callable[[SignalsInFrame], None], None] = None
+
+    def connect(self,
+                url: str,
+                api_key: Union[str, None] = None, ):
```

Comment:
```suggestion
                api_key: Union[str, None] = None):
```
Unless we want the ): on a newline for smaller diffs in the future

Reply:
I think my IDE adds these commas... but not sure. Will test

Reply:
It did not :-)